File: OPENAPI-REST-API/openapi-client/php-dt/src/App/DTO/CpuWithDefaults.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

class CpuWithDefaults
{
    #[DTA\Data(field: "id", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $id = null;

    #[DTA\Data(field: "short_desc", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $short_desc = null;

    #[DTA\Data(field: "long_desc", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $long_desc = null;

    #[DTA\Data(field: "type", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $type = null;

    #[DTA\Data(field: "speed", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $speed = null;

    #[DTA\Data(field: "num_cores", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $num_cores = null;

    #[DTA\Data(field: "num_cpus", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $num_cpus = null;

    #[DTA\Data(field: "benchmark", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $benchmark = null;

    #[DTA\Data(field: "monthly_price", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "float"])]
    public float|null $monthly_price = null;

    #[DTA\Data(field: "monthly_price_display", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $monthly_price_display = null;

    #[DTA\Data(field: "max_ram", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $max_ram = null;

    #[DTA\Data(field: "min_ram", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $min_ram = null;

    #[DTA\Data(field: "max_lff", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $max_lff = null;

    #[DTA\Data(field: "max_sff", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $max_sff = null;

    #[DTA\Data(field: "max_nve", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $max_nve = null;

    #[DTA\Data(field: "visible", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $visible = null;

    #[DTA\Data(field: "active", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $active = null;

    #[DTA\Data(field: "memory_det", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\MemoryOption::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\MemoryOption::class])]
    public \App\DTO\MemoryOption|null $memory_det = null;

    #[DTA\Data(field: "hd_det", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\HardDrive::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\HardDrive::class])]
    public \App\DTO\HardDrive|null $hd_det = null;

    #[DTA\Data(field: "monthly_fee", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $monthly_fee = null;

}

File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/CpuWithDefaults.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CpuWithDefaults._

case class CpuWithDefaults (
  id: Option[Integer],
short_desc: Option[String],
long_desc: Option[String],
`type`: Option[String],
speed: Option[String],
num_cores: Option[String],
num_cpus: Option[String],
benchmark: Option[String],
monthly_price: Option[BigDecimal],
monthly_price_display: Option[String],
max_ram: Option[String],
min_ram: Option[String],
max_lff: Option[String],
max_sff: Option[String],
max_nve: Option[String],
visible: Option[String],
active: Option[String],
memory_det: Option[MemoryOption],
hd_det: Option[HardDrive],
monthly_fee: Option[String])

object CpuWithDefaults {
  import DateTimeCodecs._

  implicit val CpuWithDefaultsCodecJson: CodecJson[CpuWithDefaults] = CodecJson.derive[CpuWithDefaults]
  implicit val CpuWithDefaultsDecoder: EntityDecoder[CpuWithDefaults] = jsonOf[CpuWithDefaults]
  implicit val CpuWithDefaultsEncoder: EntityEncoder[CpuWithDefaults] = jsonEncoderOf[CpuWithDefaults]
}
